tests/unit/server/test_standalone.py

```diff
@@ -192,11 +192,15 @@ def test_only_forking_not_limited(self, mock_cpu_count, mock_process):
             self.assertFalse(server_getter.return_value.main.called)
 
             self.assertEqual(10, mock_process.call_count)
+
+            found_ids = [0 for _ in range(0, 11)]
             i = 1
             for call in mock_process.call_args_list:
+                id = call[1]['args'][0]
                 self.assertEqual(server_getter.return_value.main, call[1]['target'])
-                self.assertEqual('pysoa-worker-{}'.format(i), call[1]['name'])
-                self.assertEqual((i, ), call[1]['args'])
+                self.assertEqual('pysoa-worker-{}'.format(id), call[1]['name'])
+                assert found_ids[id] == 0, "Already seen process id"
```

Comment:
So we created 10 processes.
This section verifies that we waited for the 10 to finish.
We also verify that they are pointing to the same target, that they follow a naming convention and that our call_args_list contains a given process id only once.

Reply:
From what I can see `found_ids` ends up being `[0, 0, 0, 0, 0, 0, 0, 0, 0, 0]`. So all process would have the same id?

Reply:
No, the initial value of found_ids is [0, 0, 0, 0, 0, 0, 0, 0, 0, 0].
As we look and examine the information in 'call' we update the index matching the id in the list.
So example: [0, 1, 0, 0, 0, 0, 0, 0, 0, 0]. If for some reason the process is in call 2 times. Then the next time we check we will see that founds_ids[1] is not 0 and we will show the error "Already seen process id".

The test was failing because it assumed that all processes with end in the sequence that they were started.
The code change ensures that we just check that all services have ended and have the naming conventions that we expect.

Reply:
Mmm it's a flag system check. Ok, I would recommend using [assert_has_calls](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_has_calls) instead in a declarative fashion so it's easier to read.